### PR TITLE
feat(emails): Send `New Device Login` email on all new logins

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We currently only send the new device emails for relying parties that request encryption keys (ie Sync), we now want to send it for all new logins
- I considered renaming the term `New Device Login` to something a bit more generic since this could now include services, but differed since that would  increase the scope alot

## This pull request

- Updates the code to send on all new logins

## Issue that this pull request solves

Closes:  https://mozilla-hub.atlassian.net/browse/FXA-5754

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Screen Shot 2022-09-01 at 11 56 18 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "format": "yarn workspaces foreach run format",
     "ports": "pm2 jlist | json -a -c 'this.pm2_env.env.PORT' pm2_env.env.PORT name",
     "heroku-postbuild": "yarn workspaces foreach --verbose --include 123done install",
-    "mysql": "docker exec -it $(docker container ls | grep mysql | cut -d' ' -f1) mysql"
+    "mysql": "docker exec -it $(docker container ls | grep mysql | cut -d' ' -f1) mysql",
+    "firefox": "./packages/fxa-dev-launcher/bin/fxa-dev-launcher.mjs"
```
Comment (vbudhram) [inline package.json:21]: Added a shortcut to launch firefox with `yarn firefox`. You can still use `npm start firefox`

File: packages/fxa-auth-server/lib/routes/account.ts [hunk 2]
```diff
@@ -1011,50 +1011,48 @@ export class AccountHandler {
...
-              this.log.trace(
-                'Account.login.sendNewDeviceLoginNotification.error',
-                {
-                  error: err,
-                }
-              );
-            }
+          const geoData = request.app.geo;
+          const service =
+           (request.payload as any).service || request.query.service;
```
Comment (dschom) [inline packages/fxa-auth-server/lib/routes/account.ts:1021]: Previously when requestHelper.wantsKey was in use, it would ensure that request.query wasn't undefined/null. Perhaps request.query still needs to be checked?

File: packages/fxa-auth-server/test/remote/account_signin_verification_tests.js [hunk 3]
```diff
@@ -331,6 +331,10 @@ describe('remote account signin verification', function () {
       .then((c) => {
         client2 = c;
       })
+      .then(() => {
+        // Clears inbox of new signin email
+        return server.mailbox.waitForEmail(email);
```
Comment (dschom) [inline packages/fxa-auth-server/test/remote/account_signin_verification_tests.js:336]: If you don't do this what happens? Does it mess up the proceeding 'waitForCode' call?

File: packages/fxa-content-server/tests/functional/settings/recovery_key.js [hunk 4]
```diff
@@ -304,7 +304,7 @@ registerSuite('Recovery key - unverified session', {
           .then(click(selectors.RECOVERY_KEY.GENERATE_KEY_BUTTON))
           // if the session is unverified, then the modal will be shown.
           .then(testElementExists('[data-testid=modal-verify-session]'))
-          .then(fillOutVerificationCode(email, 0))
+          .then(fillOutVerificationCode(email, 1))
```
Comment (dschom) [inline packages/fxa-content-server/tests/functional/settings/recovery_key.js:307]: Add a comment why we look for index 1 now and/or also check for new device login email?

File: packages/fxa-payments-server/src/lib/hooks.test.tsx [hunk 5]
```diff
@@ -209,8 +209,8 @@ describe('useInfoBoxMessage', () => {
     const couponDurationDate = getByTestId('couponDurationDate').textContent;
     expect(messageText).toBe(CouponInfoBoxMessageType.Repeating);
     expect(couponDurationDate).not.toBeNull();
-    expect(couponDurationDate?.slice(0, 8)).toBe(
-      expectedCouponDurationDate.slice(0, 8)
+    expect(couponDurationDate?.slice(0, 7)).toBe(
```
Comment (dschom) [inline packages/fxa-payments-server/src/lib/hooks.test.tsx:212]: Why is this needed? Should this be a separate fix?

**Comment by dschom** [comment]
@vbudhram I just noticed that @LZoog [noted that story book documentation should also be updated](https://mozilla-hub.atlassian.net/browse/FXA-5754?focusedCommentId=586059). Not sure what this looks like. Perhaps it should be done here or made into a separate issue so we don't lose track?

File: packages/fxa-payments-server/src/lib/hooks.test.tsx (hunk 5 above)
Comment (pdehaan) [reply packages/fxa-payments-server/src/lib/hooks.test.tsx:212]: Looks like this is from https://github.com/mozilla/fxa/pull/14069/files#diff-0d3612ed12d37aac060862c13f9face9d622f87013ed430f4c09fe83077b9ba5?

File: packages/fxa-auth-server/lib/routes/account.ts (hunk 2 above)
Comment (vbudhram) [reply packages/fxa-auth-server/lib/routes/account.ts:1021]: I think this is probably ok since it goes through joi validation.

File: packages/fxa-auth-server/test/remote/account_signin_verification_tests.js (hunk 3 above)
Comment (vbudhram) [reply packages/fxa-auth-server/test/remote/account_signin_verification_tests.js:336]: Yea it would since an extra email is now sent.

**Review by dschom — APPROVED** [review]
LGTM! I just finished some exploratory testing and things seem to be working as expected.
 
There was one use case I wanted to run by you. If I was not signed into 123done yet, but currently signed into sync, and then proceeded to sign into 123done, the `New sign-in to 123Done` email would not be sent. I assume this is okay, because there’s already an email from `New sign-in to Firefox`, but thought I’d double check.

**Comment by vbudhram** [comment]
> I assume this is okay,

I think this is ok. Behind the scenes it is using the existing sync session token to login.
